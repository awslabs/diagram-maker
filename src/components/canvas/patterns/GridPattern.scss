.dm-grid {
  shape-rendering: geometricPrecision;

  pattern {
    path {
      fill: none;
      stroke: #d5dbdb;
    }
  }
}
