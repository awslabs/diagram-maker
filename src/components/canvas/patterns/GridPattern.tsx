import * as Preact from 'preact';

export interface GridPatternProps {
  cellSize: number;
  patternSize: number;
}

const GridPattern = (props: GridPatternProps): JSX.Element => {
  const { cellSize, patternSize } = props;

  return (
    <svg className="dm-grid" xmlns="http://www.w3.org/2000/svg">
      <defs>
        <pattern id="dm-grid-smallpattern" width={cellSize} height={cellSize} patternUnits="userSpaceOnUse">
          <path d={`M ${cellSize} 0 L 0 0 0 ${cellSize}`} stroke-width="0.5" />
        </pattern>
        <pattern id="dm-grid-pattern" width={patternSize} height={patternSize} patternUnits="userSpaceOnUse">
          <rect width={patternSize} height={patternSize} fill="url(#dm-grid-smallpattern)" />
          <path d={`M ${patternSize} 0 L 0 0 0 ${patternSize}`} stroke-width="1" />
        </pattern>
      </defs>
      <rect fill="url(#dm-grid-pattern)" width="100%" height="100%" />
    </svg>
  );
};

export default GridPattern;
