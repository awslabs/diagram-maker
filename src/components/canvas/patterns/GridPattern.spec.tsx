import * as Preact from 'preact';
import { shallow } from 'preact-render-spy';

import GridPattern from './GridPattern';

describe('GridPattern', () => {
  it('renders grid pattern', () => {

    const gridPattern = shallow(
      <GridPattern cellSize={10} patternSize={100}/>
    );

    expect(gridPattern).toMatchSnapshot();
  });
});
