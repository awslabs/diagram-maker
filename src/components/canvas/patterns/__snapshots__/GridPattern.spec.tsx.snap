// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridPattern renders grid pattern 1`] = `
preact-render-spy (1 nodes)
-------
<svg
  class="dm-grid"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <pattern
      id="dm-grid-smallpattern"
      width={10}
      height={10}
      patternUnits="userSpaceOnUse"
    >
      <path
        d="M 10 0 L 0 0 0 10"
        stroke-width="0.5"
      >
      </path>
    </pattern>
    <pattern
      id="dm-grid-pattern"
      width={100}
      height={100}
      patternUnits="userSpaceOnUse"
    >
      <rect
        width={100}
        height={100}
        fill="url(#dm-grid-smallpattern)"
      >
      </rect>
      <path
        d="M 100 0 L 0 0 0 100"
        stroke-width="1"
      >
      </path>
    </pattern>
  </defs>
  <rect
    fill="url(#dm-grid-pattern)"
    width="100%"
    height="100%"
  >
  </rect>
</svg>

`;
